source/hp/dof_handler.cc

```diff
@@ -609,7 +617,7 @@ namespace internal
               types::global_dof_index cache_size = 0;
               for (typename HpDoFHandler<dim,spacedim>::active_cell_iterator
                    cell=dof_handler.begin_active(level);
-                   cell!=dof_handler.end_active(level); ++cell)
+                   cell!=end_active_cache[level]; ++cell)
```

Comment:
That's what we seem to be doing elsewhere, using an `endc` variable. It also has the advantage that you don't actually have to define and build the cache, you just hoist the constant expression out of the loop.